src/app/changelog/changelog.service.ts

```diff
@@ -0,0 +1,141 @@
+    map((x) => {
+      return x.filter((release: any) => !!release.date).sort(this.sortByDate);
+    })
+  );
+
+  constructor(private db: AngularFirestore, private github: GithubService) {}
+
+  updateChangelog(fromVersion?, toVersion?) {
+    this.getChangelog(fromVersion, toVersion).subscribe((release) => {
+      if (!this.github.useMocks) {
+        this.changelog.doc(release.name).set(release);
+      }
+    });
+  }
+
+  private getChangelog(fromVersion?, toVersion?) {
+    let tags: string[];
+    let currentTag: string;
+    let previousTag: string;
+    const delayed = 3000;
+
+    return this.versionsToUpdate(fromVersion, toVersion).pipe(
+      map((x) => {
+        tags = x.tags;
+        return x.versionsToUpdate;
+      }),
+      mergeMap((x) => x),
+      concatMap((x) => of(x).pipe(delay(delayed))),
+      switchMap((x: string) => {
+        const indexOfPreviousTag = tags.indexOf(x) + 1;
+        currentTag = x;
+        previousTag = tags[indexOfPreviousTag] || tags[tags.length - 1];
+
+        return forkJoin(this.github.compareCommits(previousTag, currentTag));
+      }),
+      switchMap((compares: any) => {
+        if (compares[0].commits.length === 0) return of([]);
+
+        const pullRequests = compares[0].commits.map((compare: any, index: number) => {
+          return of(compares).pipe(
+            delay(index * 3000),
+            switchMap(() => this.github.searchAssociatedPullRequest(compare.sha))
+          );
+        });
+        return forkJoin(pullRequests);
+      }),
+      map((searchResults: any) => {
+        let pullRequests = [];
+        let items = searchResults
+          .map((x: any) => x.items[0])
+          .filter((x: any) => {
+            return !!x;
+          });
+        return this.getUnique(items, 'number');
+      }),
+      map((pullRequests) => {
+        if (pullRequests.length === 0) return { name: currentTag };
+        return {
+          name: currentTag,
+          date: pullRequests[0].closed_at,
+          compareLink: `https://github.com/kirbydesign/designsystem/compare/${previousTag}...${currentTag}`,
+          associatedPullRequests: pullRequests.map((pullRequest) => {
+            return {
+              closed_at: pullRequest.closed_at,
+              number: pullRequest.number,
+              title: pullRequest.title,
+              labels: pullRequest.labels.map((label: any) => {
+                return { name: label.name, color: label.color };
+              }),
+            };
+          }),
+        };
+      })
+    );
+  }
+
+  private getUnique(arr, comp) {
+    const unique = arr
+      .map((e) => e[comp])
+
+      // store the keys of the unique objects
+      .map((e, i, final) => final.indexOf(e) === i && i)
+
+      // eliminate the dead keys & store unique objects
+      .filter((e) => arr[e])
+      .map((e) => arr[e]);
+
+    return unique;
+  }
+
+  private versionsToUpdate(fromVersion?, toVersion?): Observable<any> {
```

Comment:
Consider specifying types explicitly
```suggestion
  private versionsToUpdate(fromVersion?: string, toVersion?: string): Observable<any> {
```